src/Middleware/OperationsMiddleware.php

```diff
@@ -66,11 +65,14 @@ public function __invoke(Call $call, array $options)
         return $next(
             $call,
             $options
-        )->then(function (Operation $response) {
+        )->then(function (Message $response) {
             $options = $this->descriptor + [
                 'lastProtoResponse' => $response
             ];
-            return new OperationResponse($response->getName(), $this->client, $options);
+            $operationNameMethod = isset($options['operationNameMethod'])
+                ? $options['operationNameMethod'] : 'getName';
+            $operationName = call_user_func([$response, $operationNameMethod]);
```

Comment:
Why use `call_user_func` here and a "dynamic" variable function earlier (`$request->$additionalArgsMethodName()`)? Is there a different between the two? My PHP n00b eyes like the variable function style a little better, but not sure if it makes a difference.

Reply:
They are identical!

The only reason I use `call_user_func` and `call_user_func_array` in `OperationResponse` is because they're using `$this->variableName`, which can't be called with the syntax mentioned here. I wanted to be consistent so I used it again here, but forgot about `$request->$additionalArgsMethodName()`. 

I'm ambivalent, I do not think there's any difference other than readability.

```suggestion
            $operationName = $response->$operationNameMethod();
```

Reply:
Oh ok the `$this_>variableName` syntax being a "limiting factor" is new to me. TIL. Thanks, feel free to leave it all the way it is, just practicing my PHP code review sk1lLz 😋